assignments/this.js

```diff
@@ -1,26 +1,63 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. Window/Global Object Binding: When in the global scope, the value of “this” will be the window/console Object.
+* 2. Implicit Binding - 'this' refers to whatever is left of the dot at call time. For example, if you have an Object called myObject, and you were wanting to call a function on it (myObject.function), 'this' applies to whatever proceeds the dot, which would be myObject.
+* 3. New Binding - Whenever an new object is created and returned through a constructor function, ‘this’ refers to that specific object.
+* 4. Explicit Binding - is used with functions, where you can use .call, .apply, or .bind to specifically point to where 'this' refers to.
 *
 * write out a code example of each explanation above
 */
 
 // Principle 1
-
 // code example for Window Binding
+function hello (name){
+    return name;
+    console.log (this);
```

Comment:
```suggestion
    console.log (this);
```
This line will never run. Once the return statement is run, the function is stopped. Move this line up above the return statement